backend/tests/v5/unit/models/users.test.js

```diff
@@ -72,7 +72,67 @@ const testGetFavourites = () => {
 	});
 };
 
+const testAppendFavourites = () => {
+	const favouritesData = {
+		teamspace1: ['a', 'b', 'c'],
+		teamspace2: ['d'],
+	};
+
+	describe('Add containers to favourites', () => {
+		test('Adding an array of containers of a new teamspace', async () => {
+			jest.spyOn(db, 'findOne').mockResolvedValue({ customData: { starredModels: favouritesData } });
+			await User.appendFavourites('xxx', 'teamspace3', ['e', 'f']);
+			const newFavourtiesData = favouritesData;
+			newFavourtiesData.teamspace3 = ['e', 'f'];
+			expect(favouritesData).toEqual(newFavourtiesData);
+		});
+
+		test('Adding an array of containers of an existing teamspace', async () => {
+			jest.spyOn(db, 'findOne').mockResolvedValue({ customData: { starredModels: favouritesData } });
+			await User.appendFavourites('xxx', 'teamspace1', ['d', 'e']);
+			const newFavourtiesData = favouritesData;
+			newFavourtiesData.teamspace1 = ['d', 'e'];
+			expect(favouritesData).toEqual(newFavourtiesData);
+		});
+
+		test('Adding an array of containers of an existing teamspace with duplicate entries', async () => {
+			jest.spyOn(db, 'findOne').mockResolvedValue({ customData: { starredModels: favouritesData } });
+			await User.appendFavourites('xxx', 'teamspace1', ['a', 'b', 'c', ' d', 'e']);
+			const newFavourtiesData = favouritesData;
+			newFavourtiesData.teamspace1 = ['d', 'e'];
+			expect(favouritesData).toEqual(newFavourtiesData);
```

Comment:
missing tests:
1. starredModels is undefined (this should append as normal and don't throw an error)
2. user doesn't exist (this should throw error)